src/storage/src/statistics.rs

```diff
@@ -424,6 +441,13 @@ impl StorageStatistics<SourceStatisticsUpdate, SourceStatisticsMetrics> {
         cur.1.envelope_state_count = value;
         cur.2.envelope_state_count.set(value);
     }
+
+    /// Set the `rehydration_latency_seconds` to the given value.
+    pub fn set_rehydration_latency_seconds(&self, value: u64) {
+        let mut cur = self.stats.borrow_mut();
+        cur.1.rehydration_latency_seconds = Some(value);
+        cur.2.rehydration_latency_seconds.set(value);
```

Comment:
does this mean the gauge goes from `0` to the number? thats nice for `upsert`, but for non-`upsert`, it will just stay `0`, right? we might want to clarify this someday (not in this pr)

Reply:
Yeah, for non-upsert it will stay at `0`.

Reply:
OOC do you think it will be possible to create the gauge lazily? So it just won't be available for non-upsert cases? Or will that be confusing for users?

Reply:
the only users are our team, so we can try this out and change it if we want!

Reply:
sounds good!